frontend/src/app/main/data/workspace/selection.cljs

```diff
@@ -178,29 +178,58 @@
       (let [objects (wsh/lookup-page-objects state)]
         (rx/of (dwc/expand-all-parents ids objects))))))
 
+
+
+(defn- select-siblings
+  [state parent]
+  (let [children (wsh/lookup-shapes state (:shapes parent))
+        selected (into (d/ordered-set)
+                       (comp (remove :blocked) (map :id))
+                       children)]
+    (rx/of (select-shapes selected))))
+
+(defn- select-all-frame
+  [state]
+  (let [focus (:workspace-focus-selected state)
+        objects  (-> (wsh/lookup-page-objects state)
+                     (cp/focus-objects focus))
+
+        selected (let [frame-ids (into #{} (comp
+                                            (map (d/getf objects))
+                                            (map :frame-id))
+                                       (wsh/lookup-selected state))
+                       frame-id  (if (= 1 (count frame-ids))
+                                   (first frame-ids)
+                                   uuid/zero)]
+                   (cph/get-immediate-children objects frame-id))
+
+        selected (into (d/ordered-set)
+                       (comp (remove :blocked) (map :id))
+                       selected)]
+
+    (rx/of (select-shapes selected))))
+
+
 (defn select-all
   []
   (ptk/reify ::select-all
     ptk/WatchEvent
     (watch [_ state _]
-      (let [focus (:workspace-focus-selected state)
-            objects  (-> (wsh/lookup-page-objects state)
-                         (cp/focus-objects focus))
-
-            selected (let [frame-ids (into #{} (comp
-                                                (map (d/getf objects))
-                                                (map :frame-id))
-                                           (wsh/lookup-selected state))
-                           frame-id  (if (= 1 (count frame-ids))
-                                       (first frame-ids)
-                                       uuid/zero)]
-                       (cph/get-immediate-children objects frame-id))
-
-            selected (into (d/ordered-set)
-                           (comp (remove :blocked) (map :id))
-                           selected)]
-
-        (rx/of (select-shapes selected))))))
+           (let [current-selection-parents (->> (wsh/lookup-selected state)
+                                                (wsh/lookup-shapes state)
+                                                (map :parent-id)
+                                                distinct)
```

Comment:
I think this can be better expressed in this way:

```clojure
 (->> (wsh/lookup-selected state)
      (wsh/lookup-shapes state)
      (into #{} (map :parent-id)))
```

Because this will generate a set, and is countable, the map + disctinct generates a seq where the count scales O(n)

Reply:
Changed